src/cpp/session/SessionAsyncRProcess.cpp

```diff
@@ -244,8 +246,16 @@ void AsyncRProcess::onStderr(const std::string& output)
 
 bool AsyncRProcess::onContinue()
 {
+   // if this process has been interrupted, but appears to still be alive,
+   // then forcefully terminate the process after 3 seconds
    if (terminationRequested_)
-      return false;
+   {
+      auto elapsed = std::chrono::steady_clock::now() - terminationRequestedTime_;
+      if (elapsed >= std::chrono::seconds(3))
```

Comment:
We might want to increase this timeout in case we want to give exiting processes more time to finish cleaning up.